src/Models/ShallowWaterModels/ShallowWaterModels.jl

```diff
@@ -0,0 +1,18 @@
+module ShallowWaterModels
+
+using KernelAbstractions: @index, @kernel, Event, MultiEvent
+using KernelAbstractions.Extras.LoopInfo: @unroll
+
+using Oceananigans.Utils: launch!
+
+#####
+##### ShallowWaterModel definition
+#####
+
+include("shallowwater_model.jl")
```

Comment:
Perfect!

Our convention is to use underscores between words. So I think we'd recommend naming this file `shallow_water_model.jl`.

Reply:
Good suggestion and happy to make this change tomorrow.  If you have any suggestions on how to fix my error I am happy to fix that too.

Reply:
I updated the code to change shallowwater_model.jl to shallow_water_model.jl, to follow convention.

I also inserted the line that I think should pass in the layer_depth, but doesn't since it seems to only accept velocities and tracers.  I suspect this is from something deeper in Oceananigans and I don't know where that is.  

My bandaid fix will be to put the height into the tracers and hope that someone can help me clean this up later.

Reply:
The zero-dimensional (zeroD) problem seems to do something without crashing until I go to actually run it.

Simulations/run.jl fails because, I believe, it has IncompressibleModel stuff hard coded into it.    See the two lines below.  

Do people agree that these need to be updated to allow for ShallowWaterModel before I can proceed to run a simulation?

```
ab2_or_rk3_time_step!(model::IncompressibleModel{<:QuasiAdamsBashforth2TimeStepper}, Δt; euler) = time_step!(model, Δt, euler=euler)
ab2_or_rk3_time_step!(model::IncompressibleModel{<:RungeKutta3TimeStepper}, Δt; euler) = time_step!(model, Δt)

Reply:
Change `IncompressibleModel` to `AbstractModel` to get those lines to work.

But how about just focusing on getting the model to instantiate, and to the point that you can use `set!`? We can get the timestepping to work in a future PR.

Reply:
Good idea, I will do that first and we can come back to this later, after I have somethng working, which is not currently the case.